deploy/s3.sh

```diff
@@ -0,0 +1,189 @@
+#!/bin/bash
+
+#Here is the script to deploy the cert to your s3 bucket.
+#export S3_BUCKET=acme
+#export S3_REGION=eu-central-1
+#export AWS_ACCESS_KEY_ID=exampleid
+#export AWS_SECRET_ACCESS_KEY=examplekey
+
+# Checks to see if awscli present
+# If not, use curl + aws v4 signature to upload object
+# Make sure your keys have access to upload objects.
+# Also make sure your default region is correct, otherwise, override with $S3_REGION
+
+# IMPORTANT
+# If you're using acme.sh on a mac, make sure you have the latest version of openssl
+# Upgrade easily with homebrew: brew install openssl
+
+########  Public functions #####################
+
+#domain keyfile certfile cafile fullchain
+s3_deploy() {
+  _cdomain="$1"
+  _ckey="$2"
+  _ccert="$3"
+  _cca="$4"
+  _cfullchain="$5"
+  
+  if [ -z "$S3_BUCKET" ] ; then
+    _err "You haven't specified the bucket name yet."
+    _err "Please set it via export and try again."
+    _err "e.g. export S3_BUCKET=acme"
+    return 1
+  fi
+
+  if ! command -v aws; then
```

Comment:
do not use `command -v`,   we have a function `_exists()`,  it can detect if a command exists.